src/sentry/replays/lib/event_linking.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import annotations
+
+import uuid
+from hashlib import md5
+from typing import TYPE_CHECKING, Literal, TypedDict
+
+from sentry.utils import json
+
+if TYPE_CHECKING:
+    from sentry.eventstore.models import BaseEvent
+
+
+class EventLinkKafkaMessage(TypedDict):
+    type: str
+    start_time: int
+    replay_id: str
+    project_id: int
+    segment_id: None
+    payload: list[int]
+    retention_days: int
+
+
+class EventLinkPayloadIds(TypedDict, total=False):
+    debug_id: str
+    info_id: str
+    warning_id: str
+    error_id: str
+    fatal_id: str
+
+
+class EventLinkPayload(EventLinkPayloadIds):
+    type: str
+    replay_id: str
+    timestamp: int
+    event_hash: str
+
+
+def get_level_key(
```

Comment:
if anyone has better ideas on how to do this typing of a dynamic dictionary key more elegantly, let me know cc @getsentry/python-typing